From 5af291bfc12e945276f1b366e354fb7f0d8ce95b Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Wed, 28 Nov 2012 21:34:29 +0000
Subject: [PATCH 04/15] mmc: Add an mmcsilent option

When using MMC commands for dfu there is considerable noise.
Suppress mmc messages when mmcsilent is set.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 common/cmd_mmc.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/common/cmd_mmc.c b/common/cmd_mmc.c
index 7dacd51..507ceef 100644
--- a/common/cmd_mmc.c
+++ b/common/cmd_mmc.c
@@ -277,8 +277,9 @@ static int do_mmcops(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 			return 1;
 		}
 
-		printf("\nMMC %s: dev # %d, block # %d, count %d ... ",
-				argv[1], curr_device, blk, cnt);
+		if (getenv("mmcsilent") == NULL)
+			printf("\nMMC %s: dev # %d, block # %d, count %d ... ",
+					argv[1], curr_device, blk, cnt);
 
 		mmc_init(mmc);
 
@@ -300,8 +301,9 @@ static int do_mmcops(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 			BUG();
 		}
 
-		printf("%d blocks %s: %s\n",
-				n, argv[1], (n == cnt) ? "OK" : "ERROR");
+		if (getenv("mmcsilent") == NULL)
+			printf("%d blocks %s: %s\n",
+					n, argv[1], (n == cnt) ? "OK" : "ERROR");
 		return (n == cnt) ? 0 : 1;
 	}
 
-- 
1.8.1.4

